src/tapis-ui/_common/GenericModal/GenericModal.tsx

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import { Modal, ModalHeader, ModalBody, ModalFooter} from 'reactstrap';
+import { Icon } from '..';
+import styles from './GenericModal.module.scss'
+
+type ModalProps = {
+  isOpen: boolean;
+  toggle: () => void;
+  title: string;
+  body: any;
+  footer?: React.PropsWithChildren<{}>;
+};
+
+const GenericModal: React.FC<ModalProps> = ({
+  isOpen = false,
+  toggle,
+  title,
+  body,
+  footer,
+}) => {
+  return (
+    <Modal isOpen={isOpen} toggle={toggle}>
+      <ModalHeader>
```

Comment:
Refer to here for how to have the ModalHeader provide and render its own close button:

https://github.com/TACC/Core-Portal/blob/7048166309354809d7704062529a6141f543892e/client/src/components/DataFiles/DataFilesModals/DataFilesPreviewModal.js#L60

Reply:
added